.github/workflows/clang-format.yaml

```
@@ -1,43 +1,51 @@
 name: clang-format
 
 on:
-  push:
-    branches: [ master ]
-    paths:
-      - 'android/app/src/**.java'
-      - '.clang-format'
-      - '.github/workflows/clang-format.yml'
   pull_request:
     branches: [ master ]
-    paths:
-      - 'android/app/src/**.java'
+    paths:  # Should stay in sync with tools/unix/clang-format.sh
+      - '.clang-format-ignore'
       - '.clang-format'
       - '.github/workflows/clang-format.yml'
+      - 'android/app/src/**.java'
+      - 'android/sdk/src/**.java'
+      - 'android/sdk/src/main/cpp/**.[ch]pp'
+      - 'dev_sandbox/**.[ch]pp'
+      - 'generator/**.[ch]pp'
+      - 'iphone/**.[ch]pp'
+      - 'iphone/**.[hm]'
+      - 'iphone/**.mm'
+      - 'libs/**.[ch]pp'
+      - 'libs/**.[hm]'
+      - '!libs/indexer/drules_struct.pb.h'
+      - 'libs/**.mm'
+      - 'qt/**.[ch]pp'
+      - 'qt/**.h'
+      - 'tools/**.[ch]pp'
```

```suggestion
      - 'android'
      - 'dev_sandbox'
      - 'generator'
      - 'iphone'
      - 'libs'
      - 'qt'
      - 'tools'
```

This is a list that will trigger workflow
The workflow is pretty fast
Even if some unneeded files trigger it, clang-format won't check them